mesonbuild/dependencies/base.py

```diff
@@ -106,6 +107,9 @@ def _process_include_type_kw(cls, kwargs: T.Dict[str, T.Any]) -> str:
         return kwargs['include_type']
 
     def __init__(self, type_name: DependencyTypeName, kwargs: T.Dict[str, T.Any]) -> None:
+        # This allows two Dependencies to be compared even after being copied.
+        # The purpose is to allow the name to be changed, but still have a proper comparison
+        self.__id = id(self)
```

Comment:
When do we need to know 2 deps are the same? I guess that's something you spotted from CI? If not then we should add a test for that.